09_employment/compute_metrics_wage_ratio_2022.do

```diff
@@ -0,0 +1,168 @@
+
+replace wage = wage* 302.702/292.655 // 2023 CPI-U is first, then 2022 CPI-U	
+
+save "mit_living_wage-2023.dta", replace
+*/
+clear
+
+/***** import QCEW data *****/
+
+cd `raw'
+
+import delimited using "2022_data.csv", numericcols(14 15 16 17 18) varnames(1) clear
+
+/* keep only county totals */
+keep if areatype == "County" & ownership == "Total Covered"
+
+keep st cnty annualaverageweeklywage annualaverageestablishmentcount
+
+rename st state
+
+rename cnty county
+
+destring state, replace // make numeric for merging with MIT data
+
+cd `wages'
+
+/* merge living wage and QCEW data */
+merge 1:m state county using mit_living_wage-2022.dta
+
+*tab county state if _merge == 1*/// two AK counties missing from MIT data
+
+/* drop statewide obs */
+drop if _merge == 1
+
+/* only keep 1 adult, 2 children row */
+keep if adults == "1 Adult" & children == "2 Children"
+
+/* drop duplicates (first two counties repeated) */
+duplicates drop
+
+/* convert living hourly wage to weekly */
+gen weekly_living_wage = wage * 40
+
+/* get ratio (main metric) */
+gen average_to_living_wage_ratio = annualaverageweeklywage/weekly_living_wage
+
+/* create data quality flag 
+per discussion with Greg, >= 30 is 1, <30 is 3 */
+gen wage_ratio_quality = 1 if annualaverageestablishmentcount >= 30 & annualaverageestablishmentcount != .
+replace wage_ratio_quality = 3 if annualaverageestablishmentcount < 30 & annualaverageestablishmentcount != .
+replace wage_ratio_quality = . if annualaverageestablishmentcount == .
+
+/* test flag */
+tab wage_ratio_quality, missing
+
+/* put state and county in string with leading 0s */
+gen new_state = string(state,"%02.0f")
+drop state
+rename new_state state
+
+gen new_county= string(county,"%03.0f")
+drop county
+rename new_county county
+
+/* check ratio */
+/*sum average_to_living_wage_ratio, det
+hist average_to_living_wage_ratio */
+/*tab county state if average_to_living_wage_ratio == 0 */
+
+gen new_ratio = string(average_to_living_wage_ratio)
+drop average_to_living_wage_ratio
+rename new_ratio average_to_living_wage_ratio
+
+/* replace 0 ratio with missing and replace data quality as missing */
+replace average_to_living_wage_ratio = "NA" if average_to_living_wage_ratio == "."
+replace wage_ratio_quality = 3 if average_to_living_wage_ratio == "NA"
+
+save "wage_ratio_final_2022.dta",replace
+
+
+keep state county year average_to_living_wage_ratio wage_ratio_quality
+
+/* rename final variable here */
+
+*rename ratio_average_to_living_wage average_to_living_wage_ratio
+rename wage_ratio_quality ratio_living_wage_quality
+
+/* export files */
+
+order year state county average_to_living_wage_ratio ratio_living_wage_quality
+
+export delimited using metrics_wage_ratio_2022.csv, replace
+
+save "wage_ratio_final_2022.dta", replace
+
+** add in county name and state name
+global crosswalk "C:\Users\KMartinchek\Documents\umf-2024\mobility-from-poverty\geographic-crosswalks\data"
```

Comment:
need to move this up top so that users can update directories in one place

Reply:
Moved.